### DISCUSSION TITLE
Go down an alternate path in a conversation

### DISCUSSION THREAD

**Discussion by wurp** [cline/cline#1054] [Feature Requests]
Often when I'm working on something, I realize if I'd asked a different question or given a different instruction earlier in the conversation I would probably get better results.

If we could select a previous prompt in the current conversation and click "branch conversation" or something, then Cline could create a new conversation that keeps history up to that point, but let you develop the conversation from there.

As a bonus, this would let you go back to before you hit the context limit and ask for a summary of the conversation so far, so you could continue in a new conversation.

**Comment by carlgieringer** [comment]
I would like this feature too. I find myself frequently needing to selectively copy parts of a conversation ("Here's some background on this problem: .... Here's something you suggested: ... Here's my question: ...") and it would be a lot more convenient if I could "branch" the chat at a particular point, preserving the context up to that point.

I recently encountered the issue from the first comment: working on a large codebase with large files (I'm looking at you, Chromium browser_view.cc) I hit the context window and couldn't proceed without selectively copying parts of the of chat to a new prompt. With this feature, I could have branched the chat a level above the current one (where I was getting 'context window exceeded' errors from Claude/OpenRouter) and said: "summarize the primary details and results of this chat so that I can use it for a new chat."

(That second idea of summarizing and starting a new chat could be a feature on its own, but supporting branching would unlock it.)

**Reply by jeanesh** [reply]
I find once Cline has gone down the wrong path, there's a lot of redundant code that's needs to cleaned up. If we could give Cline a feedback mechanism to say 'yes' the issue is now fixed or the code works etc, it could offer to then go back and clean up code and refactor (based on the history), it would be awesome if it did that and I think it would improve the code base (readability , organisation etc) and help the LLM give better responses.

**Reply by siscia** [reply]
I fixed this problem by creating 2 files. `context.llm` and `$feature.llm`

In `context.llm` I write down all the high level info about the project. Whatever is important for the work you do.

In `$feature.llm` I keep track of all the information needed to develop the specific feature.

So, say I am working on creating the UI for the share component, the `share-ui.llm` file will contains something like:

```
This is the UI for the Share action of FooBar.

It is develop in React + Tailwind.

The share endpoints are defined in `app/share/endpoint.ts`

The types are defined in `app/share/types.ts`
```

I commit both files.

I have set a prompt for cline to always read and keep updated the `context.llm` file.

Then I just push the `$feature.llm` file to Cline anytime I am working on a specific feature.

I definitely use Cline to keep those files updated - so after it has finish doing whatever it is doing, it also automatically update those files.

Need to be very careful to actually update those files, because mistakes there propagate everywhere. So I got the habit to double check them more thoroughly that the code itself.

---

For those interested in sharing advice, I started a tiny linkedin group: https://www.linkedin.com/groups/13142106/
